### PR TITLE
Add tag manager to vulcan pages

### PR DESCRIPTION
**Pull request by aburke07**
## Overview
We want to use the tag manager on vulcan pages. We can copy the script code from the mono repo and render it in the <Head> tag of the page. We need to set some environment variables to make the tag manager work. We can copy the variables from the php app config files and add them to the .env.local file. These will need to be added to vercel in production.

## QA
The tag manager for google and matomo should now work. If there is no way to test this, just make sure the appropriate script tags are rendered in the deployment preview.

The tag manager should only work on /Vulcan pages.

Closes https://github.com/iFixit/ifixit/issues/48600

### CODE REVIEW

File: frontend/templates/troubleshooting/index.tsx [hunk 1]
```diff
@@ -75,11 +76,18 @@ const Wiki: NextPageWithLayout<{
...
          <link rel="canonical" href={canonicalUrl} />
+         <TagManager />
       </>
    );
 
+   console.log(
+      'matomo live url: ' +
+         process.env.NEXT_PUBLIC_MATOMO_TAG_MANAGER_CONTAINER_URL_LIVE
+   );
+
```
Comment (ianrohde) [inline frontend/templates/troubleshooting/index.tsx:87]: Should this be remove before proceeding?

Comment (aburke07) [reply frontend/templates/troubleshooting/index.tsx]: Thanks! [d53afcf](https://github.com/iFixit/react-commerce/pull/1803/commits/d53afcf9639f8410fdee5c46e23083a3f23ced52)

File: frontend/templates/troubleshooting/components/TagManager.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+         case 'Live':
+            return process.env
+               .NEXT_PUBLIC_MATOMO_TAG_MANAGER_CONTAINER_URL_LIVE;
+         case 'Staging':
+            return process.env
+               .NEXT_PUBLIC_MATOMO_TAG_MANAGER_CONTAINER_URL_STAGING;
+         case 'Dev':
+            return process.env.NEXT_PUBLIC_MATOMO_TAG_MANAGER_CONTAINER_URL_DEV;
+         default:
+            throw new Error(`Unknown environment: ${environment}`);
```
Comment (jarstelfox) [inline frontend/templates/troubleshooting/components/TagManager.tsx:52]: Minor:
```suggestion
            throw new Error(`Unknown environment: ${environment}\nPick from: Live, Staging, Dev`);
```

**Review by andyg0808 — COMMENTED** [review]
CR :+1:

**Review by andyg0808 — APPROVED** [review]

**Comment by erinemay** [comment]
The tag manager script in the console is set to true on this branch.

`gtag('config', 'G-5ZXNWJ73GK', true ? { debug_mode: true } : {});` - branch
`gtag('config', 'G-5ZXNWJ73GK', false ? { debug_mode: true } : {});` - master

```
<script id="gtag-ga4" data-nscript="afterInteractive">
         window.dataLayer = window.dataLayer || [];
         function gtag(){dataLayer.push(arguments);}
         gtag('js', new Date());

         gtag('config', 'G-5ZXNWJ73GK', true ? { debug_mode: true } : {});
      </script>
```

❗ It's also set this way for non-vulcan pages, like https://react-commerce-git-vulcan-tag-manager-ifixit.vercel.app/Parts/Mac_Laptop


I vaguely remember us having a way to check google tag manager in dev, but I'm not seeing any records of it on overlord or elsewhere. 

@aburke07, am I looking at the right script tag ☝️?

**Comment by jarstelfox** [comment]
@erinemay just checking you are lookign in the [new GA4 UI](https://analytics.google.com/analytics/web/?authuser=1#/a30506p383872824/admin/debugview/overview)?

**Comment by erinemay** [comment]
> @erinemay just checking you are lookign in the [new GA4 UI](https://analytics.google.com/analytics/web/?authuser=1#/a30506p383872824/admin/debugview/overview)?

I think I need permissions granted https://analytics.google.com/analytics/web/provision/?authuser=1#/provision takes me to a signup page.

**Comment by erinemay** [comment]
Hmm. It's not working for me, but I saw it working for @jarstelfox when we hopped on a call.
I disabled all my ad-blocking, refreshed, and tried on my phone and a couple different browsers. 

Example URL I'm using. 
https://react-commerce-git-vulcan-tag-manager-ifixit.vercel.app/Vulcan/Whirlpool%20Dryer%20Not%20Heating

I never saw any of my activity logged in https://analytics.google.com/analytics/web/#/a30506p383872824/admin/debugview/overview

Since I can't confirm it's working, I can't confirm that it's not happening on non-vulcan pages. I saw that debug mode is true on non-vulcan pages https://github.com/iFixit/react-commerce/pull/1803#issuecomment-1631252100.

**Comment by jarstelfox** [comment]
@erinemay do you want to try to QA this together?

**Comment by erinemay** [comment]
It's working for me now. I suspect it was the office's ip. https://github.com/iFixit/shopify-v2/pull/1065#issuecomment-1631652132

**Comment by erinemay** [comment]
It is working, 
image


but I see it logging page views for non-vulcan pages. dev_block 🦺 
image

**Comment by jarstelfox** [comment]
> but I see it logging page views for non-vulcan pages

This is expected after https://github.com/iFixit/react-commerce/pull/1759

un_dev_block 🌵

**Comment by erinemay** [comment]
> > but I see it logging page views for non-vulcan pages
> 
> This is expected after #1759
> 
> un_dev_block 🌵

qa 🦖 

deploy_block 🧷 just to be extra safe, @aburke07 were you referring to #1759 when you said this?
>The tag manager should only work on /Vulcan pages.

**Comment by sctice-ifixit** [comment]
This pull is adding [Tag Manager](https://support.google.com/tagmanager/answer/6102821?hl=en) (think of this as a tool for basically adding arbitrary JS to a page via a web UI provided by Google), which is different from the "GA4 tag", which is a specific script to send events to GA4. https://github.com/iFixit/react-commerce/pull/1759 added the GA4 tag everywhere, not tag manager.

**Comment by jarstelfox** [comment]
@sctice-ifixit thanks for the clarification. In terms of QA, I think we should follow something like this:
https://ezsegment.com/a-guide-to-gtm-preview-and-debug-mode/

[Our Tag manager is here](https://tagmanager.google.com/?authuser=1#/container/accounts/4702347565/containers/12374294/workspaces/58)

@sctice-ifixit has enough permissions to add users, I do not.

**Comment by jarstelfox** [comment]
@erinemay I removed the QA stamp and have added you to tag manager

**Comment by erinemay** [comment]
> @erinemay I removed the QA stamp and have added you to tag manager

@jarstelfox, can you invite the rest of QA so everyone can potentially test this?

**Comment by deltuh-vee** [comment]
@jarstelfox 
image
I'm not seeing the Preview button mentioned in https://ezsegment.com/a-guide-to-gtm-preview-and-debug-mode/. Do I need more permissions?

**Comment by aburke07** [comment]
Thank you @jarstelfox for getting this over the finish line!

**Pull request merged**